src/GoogleAuthenticator.php

```diff
@@ -113,26 +117,28 @@ public function getCode($secret, $time = null)
      *
      * @return string
      */
-    public function getUrl($user, $hostname, $secret)
+    public function getUrl($user, $hostname, $secret): string
     {
-        $args = func_get_args();
+        $issuer = func_get_args()[3] ?? null;
+        $label = sprintf('%s@%s', $user, $hostname);
+
         $encoder = 'https://chart.googleapis.com/chart?chs=200x200&chld=M|0&cht=qr&chl=';
-        $urlString = '%sotpauth://totp/%s@%s%%3Fsecret%%3D%s'.(array_key_exists(3, $args) && !is_null($args[3]) ? ('%%26issuer%%3D'.$args[3]) : '');
-        $encoderURL = sprintf($urlString, $encoder, $user, $hostname, $secret);
+        $urlString = 'otpauth://totp/%s%%3Fsecret%%3D%s';
+
+        if (null !== $issuer) {
+            $urlString .= '%%26issuer%%3D'.$issuer;
+        }
 
-        return $encoderURL;
+        return $encoder.sprintf($urlString, $label, $secret);
     }
 
     /**
      * @return string
      */
-    public function generateSecret()
+    public function generateSecret(): string
     {
-        $secret = random_bytes($this->secretLength);
-
-        $base32 = new FixedBitNotation(5, 'ABCDEFGHIJKLMNOPQRSTUVWXYZ234567', true, true);
-
-        return $base32->encode($secret);
+        return (new FixedBitNotation(5, 'ABCDEFGHIJKLMNOPQRSTUVWXYZ234567', true, true))
```

Comment:
Couldn't agree more with @iltar : it just puts distance between usage and definition.